### PR TITLE
Add Azure Next Gen setup info

### PR DESCRIPTION
**Pull request by lblackstone**
TODO

- [ ] Add link to Getting Started (https://github.com/pulumi/docs/issues/4295)
- [ ] Update links to tutorials

### Related issues (optional)
Fix https://github.com/pulumi/pulumi-azure-nextgen/issues/148

### CODE REVIEW

File: content/docs/intro/cloud-providers/azure-nextgen/_index.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+---
+title: Azure Next Gen
+meta_desc: The Azure Next Gen provider for Pulumi can be used to provision any of the cloud resources available in Azure via Azure Resource Manager (ARM).
+menu:
+  intro:
+    parent: cloud-providers
+    identifier: clouds-azure
```
Comment (mikhailshilkov) [inline content/docs/intro/cloud-providers/azure-nextgen/_index.md:7]:
```suggestion
    identifier: clouds-azure-nextgen
```

File: content/docs/intro/cloud-providers/azure-nextgen/_index.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+* JavaScript/TypeScript: [`@pulumi/azure-nextgen`](https://www.npmjs.com/package/@pulumi/azure-nextgen)
+* Python: [`pulumi-azure-nextgen`](https://pypi.org/project/pulumi-azure-nextgen/)
+* Go: [`github.com/pulumi/pulumi-azure-nextgen/sdk/go/azure`](https://github.com/pulumi/pulumi-azure-nextgen)
+* .NET: [`Pulumi.AzureNextGen`](https://www.nuget.org/packages/Pulumi.AzureNextGen)
+
+The Azure Next Gen provider is open source and available in the [pulumi/pulumi-azure-nextgen](https://github.com/pulumi/pulumi-azure-nextgen) repo.
+
+## Configuration
+
+The Azure provider accepts the following configuration settings.  These can be provided to the default Azure provider via `pulumi config set azure:<option>`, or passed to the constructor of `new azure.Provider` to construct a specific instance of the Azure provider.
```
Comment (mikhailshilkov) [inline content/docs/intro/cloud-providers/azure-nextgen/_index.md:64]: Right now, the settings have to be `azure-nextgen:<option>`

Comment (mikhailshilkov) [inline content/docs/intro/cloud-providers/azure-nextgen/_index.md:64]: `new azure.Provider` doesn't look right. Probably, `new azure_nextgen.Provider`?

File: content/docs/intro/cloud-providers/azure-nextgen/_index.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+* `auxiliaryTenantIds`: (Optional)
+* `clientCertificatePassword`: (Optional) The password associated with the Client Certificate. For use when authenticating as a Service Principal using a Client Certificate.
+* `clientCertificatePath`: (Optional) The path to the Client Certificate associated with the Service Principal for use when authenticating as a Service Principal using a Client Certificate.
+* `clientId`: (Optional) The Client ID which should be used.
+* `clientSecret`: (Optional) The Client Secret which should be used. For use When authenticating as a Service Principal using a Client Secret.
+* `disablePulumiPartnerId`: (Optional) This will disable the Pulumi Partner ID which is used if a custom partnerId isn’t specified.
+* `environment`: (Required) The Cloud Environment which should be used. Possible values are public, usgovernment, german, and china. Defaults to public.
```
Comment (mikhailshilkov) [inline content/docs/intro/cloud-providers/azure-nextgen/_index.md:72]: `environment` isn't required. Not sure why it's marked as required for the old provider either.

File: content/docs/intro/cloud-providers/azure-nextgen/setup.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+> **Note:** Authenticating using the CLI will not work for Service Principal logins (e.g.,
+> `az login --service-principal`).  For such cases, authenticate using the Service Principal method instead.
+
+## CLI Authentication
+
+Simply login to the Azure CLI and Pulumi will automatically use your credentials:
+
+```bash
+$ az login
+To sign in, use a web browser to open the page https://aka.ms/devicelogin and enter the code XXXFAKEXXX to authenticate.
```
Comment (mikhailshilkov) [inline content/docs/intro/cloud-providers/azure-nextgen/setup.md:33]: I believe this is changed and a device code isn't needed anymore. Could you please log in with `az login` and update the prompt?

File: content/docs/intro/cloud-providers/azure-nextgen/setup.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+Do as instructed to login.  After completed, `az login` will return and you are ready to go.
+
+For most cases `az login` should suffice, but in certain scenarios such as
```
Comment (mikhailshilkov) [inline content/docs/intro/cloud-providers/azure-nextgen/setup.md:38]: I believe this section isn't relevant anymore. The standard login should work fine for refreshing tokens. Let's remove it.

File: content/docs/intro/cloud-providers/azure-nextgen/setup.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+1. Set the environment variables `ARM_CLIENT_ID`, `ARM_CLIENT_SECRET`, `ARM_TENANT_ID`, and `ARM_SUBSCRIPTION_ID` respectively
+
+2. Set them using configuration
+
+    ```bash
+    $ pulumi config set azure:clientId <clientID>
+    $ pulumi config set azure:clientSecret <clientSecret> --secret
+    $ pulumi config set azure:tenantId <tenantID>
+    $ pulumi config set azure:subscriptionId <subscriptionId>
```
Comment (mikhailshilkov) [inline content/docs/intro/cloud-providers/azure-nextgen/setup.md:81]: `azure-nextgen` everywhere

File: content/docs/intro/cloud-providers/azure-nextgen/_index.md (hunk 3 above)
Comment (lblackstone) [reply content/docs/intro/cloud-providers/azure-nextgen/_index.md:72]: Oh, good to know. The template from the Azure getting started guide includes it, so I assumed it was required.

File: content/docs/intro/cloud-providers/azure-nextgen/_index.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+In addition to the tutorial, several interesting examples are available complete with instructions:
+
+* [Azure Function Apps](https://github.com/pulumi/examples/tree/master/azure-ts-functions): Create a serverless function
+* [Azure AppService with SQL and AppInsights](https://github.com/pulumi/examples/tree/master/azure-ts-appservice): Build an AppService web application that uses SQL and AppInsights
+* [Azure Kubernetes Service (AKS) Cluster](https://github.com/pulumi/examples/tree/master/azure-ts-aks-helm): Create an AKS cluster and deploy a Helm Chart into it
+* [Azure CosmosDB, AKS and Node.js](https://github.com/pulumi/examples/tree/master/azure-ts-aks-mean): Stands up an AKS cluster and a MongoDB-flavored instance of CosmosDB used by a Node.js application.
+
+## Example
+
+```typescript
```
Comment (lblackstone) [inline content/docs/intro/cloud-providers/azure-nextgen/_index.md:40]: TODO: turn this into a multi-language example

**Comment by mikhailshilkov** [comment]
@lblackstone Could you please add the latest changes to https://github.com/pulumi/docs/pull/5351 ? I'm planning to keep all changes in that PR from now on.

**Comment by lblackstone** [comment]
Closing out this PR in favor of #5351

**Pull request closed without merging**